util/http/http_client.cc

```diff
@@ -37,6 +38,72 @@ int VerifyCallback(int ok, X509_STORE_CTX* ctx) {
   VLOG(1) << "verify_callback: " << std::boolalpha << bool(ok);
   return ok;
 }
+
+static void ClearSslError() {
+  unsigned long l;
+
+  do {
+    const char *file, *data;
+    int line, flags;
+
+#if OPENSSL_VERSION_NUMBER >= 0x30000000
+    const char* func;
+    l = ERR_get_error_all(&file, &line, &func, &data, &flags);
+#else
+    l = ERR_get_error_line_data(&file, &line, &data, &flags);
+#endif
+  } while (l);
+}
+
+// returns -1 if failed to load any CA certificates, 0 if loaded successfully
+static int SslProbeSetDefaultCaLocation(SSL_CTX* ctx) {
```

Comment:
Should this be exposed? We have a different function that sets up a client context in `protocol_client.cc:CreateSslClientCntx()`

Reply:
would you like me to expose it via http client interface? or maybe put into tls code?

Reply:
I think a free function is fine (probably in some outside header?)